src/main/java/net/mcreator/ui/wysiwyg/WYSIWYG.java

```diff
@@ -436,6 +436,7 @@ private void drawGrid(Graphics2D g) {
 					wysiwygEditor.mcreator.getWorkspace())) {
 				if (ey >= component.getY() && ey <= component.getY() + component.getHeight(
 						wysiwygEditor.mcreator.getWorkspace())) {
+					if (component.locked) continue;

```

Comment:
This prevents the component from being selected at all in the editor. It may make sense to still allow clicking on the component and editing its properties, just locking the movement

Reply:
I did the same thing at first, then I thought that this way would be better for overlapping components.

Reply:
> Locked components can not be moved.

According to the PR description, it should only prevent movement.

I think people may still want to click on the component even if they prevent it from being moved.

You could, however, make the algorithm of "getGUIComponentAt" "prefer" (by e.g. sorting before iterating) the unlocked component over locked one, for example

Reply:
Can you check that if the mouse is clicked, it will work normally, but if the mouse is dragged, it will return the component that is not locked?
An idea came to my mind, is it possible to put another button to hide the component and the lock button only stops the movement?